pkg/utils/release/controller.go

```diff
@@ -424,3 +450,132 @@ func (s *SuiteController) CreateReleasePipelineRoleBindingForServiceAccount(name
 	}
 	return roleBinding, nil
 }
+
+func (s *SuiteController) WaitForReleasePipelineToBeFinished(managedNamespace, devNamespace string, releaseName string, component *appstudioApi.Component, maxRetries int) error {
```

Comment:
I am sure I will have more comments on this function, but I don't really know what its goal even is for now. Maybe the comment will help

Reply:
added comment, 
this function is using the `wait.pollImediate`  method to replace the `eventually` ,  trying to avoid flaky cases 
and to improve reporting  PipelineRun  name status for better debugging .

Reply:
If the main difference is replacing an eventually with a wait.pollImmediate, why is the entire function new? How was 30 mins, 20 seconds chosen?